packages/amqp/lib/AbstractAmqpConsumerMonoSchema.ts

```diff
@@ -20,28 +21,70 @@ const DEFAULT_BARRIER_RESULT = {
 
 export abstract class AbstractAmqpConsumerMonoSchema<
     MessagePayloadType extends object,
+    ExecutionContext = undefined,
+    PrehandlerOutput = undefined,
     BarrierOutput = undefined,
   >
-  extends AbstractAmqpBaseConsumer<MessagePayloadType, BarrierOutput>
+  extends AbstractAmqpBaseConsumer<
+    MessagePayloadType,
+    ExecutionContext,
+    PrehandlerOutput,
+    BarrierOutput
+  >
   implements QueueConsumer
 {
   private readonly messageSchema: ZodSchema<MessagePayloadType>
   private readonly schemaEither: Either<Error, ZodSchema<MessagePayloadType>>
+  private readonly prehandlers?: Prehandler<
+    MessagePayloadType,
+    ExecutionContext,
+    PrehandlerOutput
+  >[]
 
   constructor(
     dependencies: AMQPConsumerDependencies,
     options:
-      | (NewAMQPConsumerOptions & MonoSchemaQueueOptions<MessagePayloadType>)
-      | (ExistingAMQPConsumerOptions & MonoSchemaQueueOptions<MessagePayloadType>),
+      | (NewAMQPConsumerOptions<MessagePayloadType, ExecutionContext, PrehandlerOutput> &
+          MonoSchemaQueueOptions<MessagePayloadType>)
+      | (ExistingAMQPConsumerOptions<MessagePayloadType, ExecutionContext, PrehandlerOutput> &
+          MonoSchemaQueueOptions<MessagePayloadType>),
   ) {
     super(dependencies, options)
 
+    this.prehandlers = options.prehandlers
     this.messageSchema = options.messageSchema
     this.schemaEither = {
       result: this.messageSchema,
     }
   }
 
+  protected override processPrehandlers(message: MessagePayloadType, _messageType: string) {
+    return this.processPrehandlersInternal(this.prehandlers, message)
+  }
+
+  // MonoSchema support is going away in the next semver major, so we don't care about coverage strongly
+  /* c8 ignore start */
+  // eslint-disable-next-line max-params
```

Comment:
Nice! removing mono consumers will help to speed up development on the library 🙌, thanks Igor 🙇